lib/scala/downloader/src/main/scala/org/enso/downloader/archive/TarGzWriter.scala

```diff
@@ -0,0 +1,126 @@
+package org.enso.downloader.archive
+
+import org.apache.commons.compress.archivers.tar.{
+  TarArchiveEntry,
+  TarArchiveOutputStream
+}
+import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream
+import org.enso.cli.task.{TaskProgress, TaskProgressImplementation}
+
+import java.io.{BufferedOutputStream, FileOutputStream}
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Path}
+import scala.util.{Try, Using}
+
+/** A helper class for writing TAR archives compressed with gzip. */
```

Comment:
Is there a reason we chose gzip compression? Given we're on the hook for bandwidth costs we might want to investigate more efficient compression schemes such as zstandard or xz.

Reply:
Good point. I guess that is because .tar.gz is commonly used, but I think you are right that `xz` would work better. But I definitely cannot improve this before my time-off as this would require some more changes and doc updates. Maybe we could create a separte issue for this?

Reply:
It definitely isn't something to do now, but just something worth thinking about.

Reply:
Added a point in #1773 saying that it should be discussed.